src/main/java/manufacturing/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package manufacturing.service.impl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import manufacturing.dao.CarDao;
+import manufacturing.dao.DriverDao;
+import manufacturing.lib.Inject;
+import manufacturing.lib.Service;
+import manufacturing.model.Car;
+import manufacturing.model.Driver;
+import manufacturing.service.CarService;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+    @Inject
+    private DriverDao driverDao;
+
+    @Override
+    public Car create(Car car) {
+        return carDao.create(car);
+    }
+
+    @Override
+    public Car get(Long id) {
+        return carDao.getById(id).orElseThrow();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return carDao.getAllCars();
+    }
+
+    @Override
+    public Car update(Car car) {
+        return carDao.update(car);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.deleteById(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        driverDao.create(driver);
+        car.getDrivers().add(driver);
+        carDao.update(car);
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        car.getDrivers().remove(driver);
+        carDao.update(car);
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        return carDao.getAllCars().stream()
+                .filter(c -> c.getDrivers()
+                        .stream()
+                        .map(Driver::getId)
+                        .collect(Collectors.toList())
+                        .contains(driverId))
```

Comment:
we can simplify this block by using `anyMatch`  with predicate `car.getId().equals(driverId)` - meaning you filter only those `drivers` who have same `driverId`. This wrap up removes necessity to collect matched drivers.